### PR TITLE
Add metadata test 

### PR DESCRIPTION
**Pull request by grehx**

### CODE REVIEW

File: regression-tests/automation/metadata_test.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import unittest
+import sys
+import os
+from sparktkregtests.lib import sparktk_test
+import sparktkregtests.lib.config as config
+
+class MetadataTest(sparktk_test.SparkTKTestCase):
+
+    def test_metadata_has_no_duplicates(self):
+        "Test metadata file does not have duplicates"
```
Comment (lewisc) [inline regression-tests/automation/metadata_test.py]: make these docstrings (triple quoted)

File: regression-tests/automation/metadata_test.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        meta_list = []
+        filename = self.get_local_dataset('metadata.txt')
```
Comment (lewisc) [inline regression-tests/automation/metadata_test.py:32]: again, construct absolute paths off of config.root

File: regression-tests/automation/metadata_test.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        with open(filename, 'r') as f:
+            for line in f:
+                alpha = line.split(',', 2)
+                meta_list.append(alpha[0])
+        datasets = []
+        for file in os.listdir(config.dataset_directory):
+            datasets.append(file)
+        datasets.remove('metadata.txt')
```
Comment (lewisc) [inline regression-tests/automation/metadata_test.py:40]: config.root path

File: regression-tests/automation/metadata_test.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        duplicates = len(meta_list) - len(set(meta_list))
+
+        self.assertEqual(duplicates, 0)
+
+    def test_metadata_has_all_datasets(self):
+        "Test metadata has every dataset in the dataset directory"
+        meta_list = []
+        filename = self.get_local_dataset('metadata.txt')
```
Comment (lewisc) [inline regression-tests/automation/metadata_test.py:48]: config.root path

File: regression-tests/automation/metadata_test.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        with open(filename, 'r') as f:
+            for line in f:
+                alpha = line.split(',', 2)
+                meta_list.append(alpha[0])
+        datasets = []
+        for file in os.listdir("../datasets"):
```
Comment (lewisc) [inline regression-tests/automation/metadata_test.py]: do the same thing here as in line 38

**Review by lewisc — CHANGES_REQUESTED** [review]

File: regression-tests/automation/metadata_test.py (hunk 3 above)
Comment (lewisc) [reply regression-tests/automation/metadata_test.py:40]: whoops, nevermind on this.

File: regression-tests/automation/metadata_test.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        meta_list = []
+        filename = self.get_local_dataset('metadata.txt')
+        with open(filename, 'r') as f:
+            for line in f:
+                alpha = line.split(',', 2)
+                meta_list.append(alpha[0])
+        datasets = []
+        for file in os.listdir(config.dataset_directory):
+            datasets.append(file)
+        datasets.remove('metadata.txt')
```
Comment (blbarker) [inline regression-tests/automation/metadata_test.py:40]: I don't think you need the "datasets" computation in this function?

File: regression-tests/automation/metadata_test.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        is_not_documented = list(set(datasets) - set(meta_list))
+
+        self.assertEqual(is_not_documented, [])
+
+    def test_metadata_does_not_have_nonexistent_dataset(self):
```
Comment (blbarker) [inline regression-tests/automation/metadata_test.py:61]: This test appears to be a duplicate of the previous one.

Comment (blbarker) [reply regression-tests/automation/metadata_test.py:61]: In fact, it seems you could easily collapse all three into one test, with two asserts

Comment (grehx) [reply regression-tests/automation/metadata_test.py:61]: its easier to read the output when it errors when they're broken up, but yeah, I can collapse them

File: regression-tests/automation/metadata_test.py (hunk 6 above)
Comment (grehx) [reply regression-tests/automation/metadata_test.py:40]: yeah I'll fix that

**Pull request merged**